### PR TITLE
Additional refactoring to bolt class

### PR DESCRIPTION
**Pull request by royfalk**
Purpose:
- What is this pull request trying to do? Refactor bolt classes. Merge bolt_generic into bolt. etc.
- What release is this for? Next one.
- Is there a project or milestone we should apply this to? Not really.

Please answer the following:

Code Changes:
- [Yes] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation

Issues:
- Intermittent jump freeze for several seconds (visual only). This was already seen in previous commits.

### CODE REVIEW

File: engine/src/gfx/boltdrawmanager.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+{
+    unsigned int i;
+    for (i = 0; i < animations.size(); i++)
+        delete animations[i];
+    for (i = 0; i < balls.size(); i++)
+        for (int j = balls[i].size()-1; j >= 0; j--)
+            balls[i][j].Destroy( j );
+    for (i = 0; i < bolts.size(); i++)
+        for (int j = bolts[i].size()-1; j >= 0; j--)
+            bolts[i][j].Destroy( j );
```
Comment (BenjamenMeyer) [inline engine/src/gfx/boltdrawmanager.cpp:47]: please add braces around these

**Review by BenjamenMeyer — CHANGES_REQUESTED** [review]
a few code style changes and it should be good - it's part of updating/migrating code and should help fix bugs too.

**Comment by evertvorster** [comment]
I've given it a good play test, and everything seems to work fine, with no new issues. I play-tested with the Llama, accepted missions, and even shot at Oswald a little. 

Then I tested with the Vendetta, and I got a little carried away, and jumped from the edge of my exploratory efforts to Sol/Sol over about 20 jumps. Sol/Sol did not load on my system, and crashed the game. Of course, I have not saved my game all the way in, which is a little annoying.

**Review by evertvorster — APPROVED** [review]
No new issues to report. Looks good to me!

**Review by BenjamenMeyer — APPROVED** [review]
Thanks!

**Comment by BenjamenMeyer** [comment]
given the last commit is only whitespace changing for code style it's good for merge

**Pull request merged**